src/translate.service.ts

```diff
@@ -286,9 +286,12 @@ export class TranslateService {
      * @param interpolateParams
      * @returns {any}
      */
-    public getParsedResult(translations: any, key: any, interpolateParams?: Object): any {
+    public getParsedResult(translations: any, key: any, interpolateParams?: any): any {
         let res: string|Observable<string>;
 
+        interpolateParams = interpolateParams || {};
+        interpolateParams.currentLang = interpolateParams.currentLang || this.currentLang;
```

Comment:
This is a problem, it will define `interpolateParams` even when there are no params. Why do you need the lang in the custom plural parser ?